test/powershell/engine/DefaultCommands.Tests.ps1

```diff
@@ -528,4 +528,225 @@ Describe "Verify approved aliases list" -Tags "CI" {
     It "Should have 'more' as a function" {
         Test-Path Function:more | Should Be $true
     }
+
+    It "Should have 'sudo' as a function only on CoreUnix" {
+        Test-Path Function:sudo | Should Be $CoreUnix
+    }
+}
+
+if ($isCoreCLR -and !$IsWindows)
+{
+[string] $script:sudoTestPath = $null
+[string] $script:sudoMockPath = $null
+[string] $script:sudoSaveEnv = $null
+[string] $script:sudoMock = @"
+#!/bin/sh
+echo `$@
+"@
+}
+
+Describe "Verify sudo function on CoreUNIX" -Tags "CI"{
+
+    Context "Verify using MOCK sudo" {
+
+        BeforeAll {
+            $skipTest = ((-not $isCoreCLR) -or $IsWindows)
+            if ($skipTest) {return}
+
+            # create a test directory for the mock sudo.
+            $item = new-item -Path $testdrive -Name 'sudobin' -ItemType Directory -ErrorAction Stop
+            $script:sudoTestPath = $item.FullName
+
+            # save the previous path and prepend the test directory
+            $script:sudoSaveEnv = $env:PATH
+            $env:PATH = $sudoTestPath + ':' + $env:PATH
+
+            # place a mock sudo command in the test directory
+            $script:sudoMockPath = Join-Path -Path $script:sudoTestPath -ChildPath 'sudo'
+            $sudoscript = $script:sudoMock.Replace("`r`n", "`n")
+            Set-Content -Path $script:sudoMockPath -Value $sudoscript
+            chmod 777 $script:sudoMockPath
+
+            # Verify sudo resolves to the mock
+            $items = Get-Command -Name 'sudo' -CommandType Application
+            if ($items -eq $null)
+            {
+                throw "Could not resolve $($script:sudoMockPath) command"
+            }
+            $cmd = $null
+            if ($items.GetType().IsArray)
+            {
+                $cmd = $items[0]
+            }
+            else
+            {
+                $cmd = $items
+            }
+            if ($cmd.Path -ne $script:sudoMockPath)
+            {
+                throw "Get-Command did not resolve to the mock sudo. Expected: $($script:sudoMockPath)  Actual: $($cmd.Path)"
+            }
+        }
+
+        AfterAll {
+            if ($skipTest) {return}
+
+            # restore the environment
+            if (-not [string]::IsNullOrEmpty($script:sudoSaveEnv))
+            {
+                $env:PATH = $script:sudoSaveEnv
+            }
+
+            # clean up the mock files
+            if (Test-Path -Path $script:sudoTestPath)

```

Comment:
strictly speaking this shouldn't be needed. the contents of TESTDRIVE are cleaned up by the harness